test/Transition-test.js

```diff
@@ -469,4 +469,30 @@ describe('Transition', () => {
       wrapper.setState({ in: false })
     })
   })
+
+  describe('transition node', () => {
+    it('delegates to findDOMNode when a transntionNode is not provided', () => {
+      const findDOMNodeSpy = jest.spyOn(ReactDOM, 'findDOMNode');
+      mount(
+        <Transition in appear timeout={0}>
+          <div />
+        </Transition>
+      )
+
+      expect(findDOMNodeSpy).toHaveBeenCalledTimes(1);
+      findDOMNodeSpy.mockRestore();
+    })
+
+    it('uses transitionNode when provided', () => {
+      const onEnterSpy = jest.fn();
+      const transitionNode = document.createElement('span');
```

Comment:
But that is not the element that is transitioning. It should be the child of `Transition` which is a `div` in this case.

Reply:
should use ref